docs/pip/compile.md

```diff
@@ -60,18 +60,33 @@ $ uv pip compile pyproject.toml --all-extras
 
 Note extras are not supported with the `requirements.in` format.
 
-To lock dependency groups in the current project directory's pyproject.toml:
+To lock a dependency group in the current project directory's `pyproject.toml`, for example the
+group `foo`:
 
 ```console
 $ uv pip compile --group foo
 ```
 
-To lock dependency groups for an arbitrary pyproject.toml:
+To specify the project directory where groups should be sourced from:
 
 ```console
-$ uv pip compile --group some/path/pyproject.toml:foo
+$ uv pip compile --project some/path/ --group foo --group bar
 ```
 
+Alternatively, you can specify a path to a `pyproject.toml` for each group:
+
+```console
+$ uv pip compile --group some/path/pyproject.toml:foo --group other/pyproject.toml:bar
+```
+
+!!! note
+
+    `--group` flags do not apply to other specified sources. For instance,
+    `uv pip compile some/path/pyproject.toml --group foo` sources `foo`
+    from `./pyproject.toml` and **not** `some/path/pyproject.toml`.
+    This is an attempt to make `uv pip compile` and `uv pip install` behave similarly.
+    A `--group` flag has to be added to pip-tools' `pip compile`, [although they're considering it](https://github.com/jazzband/pip-tools/issues/2062). We expect to support whatever syntax and semantics they adopt.
```

Comment:
I would not say "This is an attempt" in public facing copy. I'd just omit that whole sentence.

I think the second bit is a separate note? I'd use `!!! note` for the first part. Then add a separate `!!! important` that says

> A `--group` flag has not yet been implemented in `pip-tools` and we may change our semantics to match the upstream behavior when it is finalized. The upstream behavior can be tracked in ...

I'd move this to the first use of `pip compile --group` above a bit so there's not too admonitions in a row.

Reply:
heh, yeah I wanted two separate notes but got stuck on the back-to-backness. adding distance between them is a fine compromise.